### ISSUE TITLE
Add underscore as a decimal separator for string formatting

### ISSUE THREAD

**Issue by 4f2e3846-682e-4d98-8367-e267de3ba0d6** [python/cpython#87790]
BPO | [43624](https://bugs.python.org/issue43624)
--- | :---
Nosy | @rhettinger, @mdickinson, @vstinner, @ericvsmith, @serhiy-storchaka, @domdfcoding

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-03-25.17:19:07.475>
labels = ['interpreter-core', 'type-feature', '3.11']
title = 'Add underscore as a decimal separator for string formatting'
updated_at = <Date 2021-05-04.15:37:03.108>
user = 'https://bugs.python.org/TerryDavis'
```

bugs.python.org fields:
```python
activity = <Date 2021-05-04.15:37:03.108>
actor = 'Terry Davis'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Interpreter Core']
creation = <Date 2021-03-25.17:19:07.475>
creator = 'Terry Davis'
dependencies = []
files = []
hgrepos = []
issue_num = 43624
keywords = []
message_count = 17.0
messages = ['389508', '389512', '389517', '389529', '389534', '389546', '389547', '389574', '389687', '389708', '389709', '389735', '389736', '389744', '389754', '389762', '392911']
nosy_count = 7.0
nosy_names = ['rhettinger', 'mark.dickinson', 'vstinner', 'eric.smith', 'serhiy.storchaka', 'Terry Davis', 'domdfcoding']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue43624'
versions = ['Python 3.11']
```

</p></details>

### Linked PRs
* gh-125304
* gh-132202
* gh-132204
* gh-132205
* gh-136364
* gh-136365

**Comment by skirpichev** [comment]
It seems, nobody is working on; I'll make a patch.

**Comment by skirpichev** [comment]
PR is ready for review: https://github.com/python/cpython/pull/125304

**Comment by skirpichev** [comment]
Gentle ping, if anyone is interested in review: https://github.com/python/cpython/pull/125304

**Comment by skirpichev** [comment]
Currently [Fraction docs](https://docs.python.org/3.13/library/fractions.html#fractions.Fraction.__format__) just point to the [format specification mini-language](https://docs.python.org/3.13/library/string.html#formatspec) page.  The Decimal type also has no special docs for formatting.

So, I suggest to implement support for digit separators in the fractional part also for Decimal's and Fraction's.

**Comment by serhiy-storchaka** [comment]
@hugovk, I think that this feature for `Fraction` and `Decimal` should be added in 3.14, as for `float`. It will just be simpler to users if all non-integer numerical types support it in the same version.

**Comment by ericvsmith** [comment]
I agree with @serhiy-storchaka that it would be good if there weren't a version where this didn't apply to all numeric types. I suggest marking this as a potential release blocker.

**Comment by hugovk** [comment]
Yes, let's include with 3.14 (if we squint, it's sort of a bug fix/omission from the first PR). I've merged https://github.com/python/cpython/pull/132202 and https://github.com/python/cpython/pull/132204.

@skirpichev Please could you take care of the backports so they'll be in tomorrow's 3.14.0b4?

And do we need any documentation updates?

**Comment by skirpichev** [comment]
> Please could you take care of the backports so they'll be in tomorrow's 3.14.0b4?

@hugovk, I can.  But maybe we can consider first backporting to 3.14 https://github.com/python/cpython/pull/130663 and https://github.com/python/cpython/pull/132549 ?

> And do we need any documentation updates?

I think - no.  All is documented in https://docs.python.org/3.14/library/string.html#format-specification-mini-language

**Comment by serhiy-storchaka** [comment]
We need the documentation updates if we _don't_ backport these changes.

**Comment by skirpichev** [comment]
I think it's done.
